src/components/SearchBar/SearchBar.tsx

```diff
@@ -89,13 +89,12 @@ const SearchBar: React.FC<AppBarProps> = ({ onDrawerToggle }) => {
   };
 
   const handleInputChange = _debounce(
-    (_: React.ChangeEvent<Record<string, unknown>>, value: string) => {
-      if (optionSelectedFromList.current || value.length < 1) return null;
-
+    (_: React.ChangeEvent<Record<string, unknown>>, _value: string) => {
+      const value = _value.replace(/\s/g, '');
+      if (optionSelectedFromList.current || !value.trim().length) return null;

```

Comment:
NB: no need to trim since all white spaces were already removed, on line 97 as well